pyrokinetics/gk_code/GKInput.py

```diff
@@ -175,6 +175,13 @@ def get_numerics(self) -> Numerics:
         """
         pass
 
+    @abstractmethod
+    def get_poincare_fac(self, dpsidr: float, rhostar: float, Lx: float) -> tuple:
```

Comment:
May as well call this get_poincare_factors

Reply:
Done!